include/dlaf/mc/cholesky_local.h

```diff
@@ -0,0 +1,90 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/parallel_executors.hpp>
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/util_matrix.h"
+
+/// @file
+
+namespace dlaf {
+
+/// @brief Cholesky implementation on local memory
+///
+/// @param uplo specifies that the matrix is \a Lower triangular
+/// @tparam mat refers to a dlaf::Matrix object
+///
+/// @throws std::runtime_error if \p uplo = \a Upper decomposition is chosen (not yet implemented)
+
+template <class T>
+void cholesky_local(blas::Uplo uplo, Matrix<T, Device::CPU>& mat) {
+  // Set up executor on the default queue with high priority.
+  hpx::threads::scheduled_executor matrix_HP_executor =
+      hpx::threads::executors::pool_executor("default", hpx::threads::thread_priority_high);
+
+  // Set up executor on the default queue with default priority.
+  hpx::threads::scheduled_executor matrix_normal_executor =
+      hpx::threads::executors::pool_executor("default", hpx::threads::thread_priority_default);
+
+  // Check if matrix is square
+  util_matrix::assert_size_square(mat, "Cholesky", "mat");
+  // Check if block matrix is square
+  util_matrix::assert_blocksize_square(mat, "Cholesky", "mat");
+
+  // Number of tile (rows = cols)
+  SizeType nrtile = mat.nrTiles().cols();
+
+  if (uplo == blas::Uplo::Lower) {
+    for (SizeType k = 0; k < nrtile; ++k) {
+      // Cholesky decomposition on mat(k,k) r/w potrf (lapack operation)
+
+      hpx::dataflow(matrix_HP_executor, hpx::util::unwrapping(tile::potrf<T, Device::CPU>), uplo,
+                    std::move(mat({k, k})));
```

Comment:
This is absolutely not wrong, it is just a suggestion.

Indices are strongly-typed with the aim to provide an additional implicit check of the index passed to a function , and brace initialization is a (legit) way to work around this check.

In order to profit from this check, I would suggest you specify the type of index you are creating, e.g. `LocalTileIndex{k, k}`.

About `{k, k}` in particular, you can also extract it in a variable and re-use it in the algorithm.

Reply:
I'll definitely follow your advice, to perform the additional implicit check of the index. Thanks!